Inline comments are not taken into account when laying out 'let'

Input:

```haskell
foo :: Int -> [Int] -> Bool
foo =
  let {-join-} go = case () of
                () -> undefined
  in go
```

Output:

```haskell
foo :: Int -> [Int] -> Bool
foo =
  let {-join-} go = case () of
        () -> undefined
   in go
```

The result is unparseable:

```
join003.hs:4:9: error: …
    parse error on input ‘(’
```

--- a/data/examples/declaration/value/function/let-multi-line-out.hs
+++ b/data/examples/declaration/value/function/let-multi-line-out.hs
@@ -10,3 +10,9 @@ bar x =
       y = x
    in z +
         100
+
+inlineComment :: Int -> Int
+inlineComment =
+  let {- join -} go = case () of
+                   () -> undefined
+   in go

--- a/data/examples/declaration/value/function/let-multi-line.hs
+++ b/data/examples/declaration/value/function/let-multi-line.hs
@@ -10,3 +10,9 @@ bar x =
       y = x
   in z
       + 100
+
+inlineComment :: Int -> Int
+inlineComment =
+  let {- join -} go = case () of
+                   () -> undefined
+  in go

--- a/src/Ormolu/Printer/Meat/Declaration/Value.hs
+++ b/src/Ormolu/Printer/Meat/Declaration/Value.hs
@@ -392,7 +392,7 @@ p_hsLocalBinds = \case
           (Left <$> bagToList bag) ++ (Right <$> lsigs)
         p_item (Left x) = located x p_valDecl'
         p_item (Right x) = located x p_sigDecl'
-    newlineSep p_item (sortOn ssStart items)
+    newlineSep (sitcc . p_item) (sortOn ssStart items)
   HsValBinds NoExt _ -> notImplemented "HsValBinds"
   HsIPBinds NoExt _ -> notImplemented "HsIPBinds"
   EmptyLocalBinds NoExt -> return ()

